### ISSUE TITLE
Renamed files have an empty Name field in FileStat

### ISSUE THREAD

**Issue by gwd** [go-git/go-git#261]
Take for example `https://xenbits.xen.org/git-http/xen.git`, commit `9872d99215ba9234b927d2c55625b70b4f555c60`.  Two of the files in that commit have been detected as a rename; the git output for one of those files looks like this:

```diff
diff --git a/xen/arch/x86/microcode.c b/xen/arch/x86/cpu/microcode/core.c
similarity index 99%
rename from xen/arch/x86/microcode.c
rename to xen/arch/x86/cpu/microcode/core.c
index 27a88c6826..ac5da6b2fe 100644
--- a/xen/arch/x86/microcode.c
+++ b/xen/arch/x86/cpu/microcode/core.c
[...]
```

The `FileStat` entry that matches that hunk has `Addition` set to 4, `Deletion` set to 11, but `Name` is the empty string.

It would be good to be able to say something usefully about renames: either make Name a `[]Name`, or (perhaps better and more backwards-compatible), add a `NewName` field which, if non-null, would indicate that `Name` was renamed to `NewName`.

BTW, thanks so much for this module -- it's been quite helpful!

**Comment by chlunde** [comment]
Was this fixed in #941 or is it about programatically accessing the old and new name?

**Comment by pjbgf** [comment]
@chlunde the PR you shared fixed this. Based on a slighly modified version of `TestStatsWithRename`:

```golang
	fileStats, err := commit.Stats()
	s.NoError(err)
	fmt.Println(fileStats[0].Name)
	// output:
	// foo => bar

	parent, err := commit.Parent(0)
	s.NoError(err)

	patch, err := commit.Patch(parent)
	s.NoError(err)

	fmt.Println(patch.String())
	// output:
	// diff --git a/bar b/foo
	// rename from bar
	// rename to foo
```

Closing as the [commit](https://github.com/go-git/go-git/commit/aebf868d2a557384539cf91476d21901ab378643) is in both `v5` releases and `main`.
